addons/account/models/account_invoice.py

```diff
@@ -383,6 +388,9 @@ def _get_sequence_prefix(self):
     @api.multi
     def _set_sequence_next(self):
         ''' Set the number_next on the sequence related to the invoice/bill/refund'''
+        user = self.env.user
+        if not user._is_admin() and not user.has_group('account.group_account_user') and not user.has_group('account.group_account_manager'):
+            return
```

Comment:
in the setter, check again that no invoice exists in the journal -> refactor the check